libc/string_errsig.c

```diff
@@ -0,0 +1,312 @@
+{
+	const int values[] = { INT_MIN, INT_MIN / 2, INT_MIN / 4, -1024, -256, -128, -64, -1, 0, 128, 256, 1024, INT_MAX / 4, INT_MAX / 2, INT_MAX };
+	for (int i = 0; i < (sizeof(values) / sizeof(values[0])); i++) {
+		errno = 0;
+		TEST_ASSERT_NOT_NULL(strsignal(signal_codes[i]));
+		TEST_ASSERT_NOT_NULL(strstr(strsignal(values[i]), "Unknown signal"));
+		TEST_ASSERT_EQUAL_INT(0, errno);
+	}
+}
+
+
+TEST(string_errsign, strsignal_real_time)
+{
+/* Disabled because of #687 issue: https://github.com/phoenix-rtos/phoenix-rtos-project/issues/687 */
+#ifdef __phoenix__
+	TEST_IGNORE_MESSAGE("#687 issue");
+#else
+	TEST_ASSERT_NOT_NULL(strsignal(SIGRTMIN));
+	TEST_ASSERT_NOT_NULL(strstr(strsignal(SIGRTMIN), "0"));
+
+	TEST_ASSERT_NOT_NULL(strsignal(SIGRTMIN + 1));
+	TEST_ASSERT_NOT_NULL(strstr(strsignal(SIGRTMIN + 1), "1"));
+
+	TEST_ASSERT_NOT_NULL(strsignal(SIGRTMIN + RTSIG_MAX / 2));
+	TEST_ASSERT_NOT_NULL(strstr(strsignal(SIGRTMIN + RTSIG_MAX / 2), "16"));
+
+	TEST_ASSERT_NOT_NULL(strsignal(SIGRTMIN + RTSIG_MAX));
+	TEST_ASSERT_NOT_NULL(strstr(strsignal(SIGRTMIN + RTSIG_MAX), "Unknown"));
+
+	TEST_ASSERT_NOT_NULL(strsignal(SIGRTMAX));
+	TEST_ASSERT_NOT_NULL(strstr(strsignal(SIGRTMAX), "30"));
+
+	TEST_ASSERT_NOT_NULL(strsignal(SIGRTMAX - 1));
+	TEST_ASSERT_NOT_NULL(strstr(strsignal(SIGRTMAX - 1), "29"));
+
+	TEST_ASSERT_NOT_NULL(strsignal(SIGRTMAX - RTSIG_MAX / 2));
+	TEST_ASSERT_NOT_NULL(strstr(strsignal(SIGRTMAX - RTSIG_MAX / 2), "14"));
+
+	TEST_ASSERT_NOT_NULL(strsignal(SIGRTMAX - RTSIG_MAX));
+	TEST_ASSERT_NOT_NULL(strstr(strsignal(SIGRTMAX - RTSIG_MAX), "Unknown"));
+#endif
+}
+
+
+TEST(string_errsign, psignal_basic)
+{
+/* Disabled because of #695 issue: https://github.com/phoenix-rtos/phoenix-rtos-project/issues/695 */
+#ifdef __phoenix__
+	TEST_IGNORE_MESSAGE("#695 issue");
+#else
+	/* Since these function shall not return a value, we check only errno */
+	FILE *err_file = fopen("stderr.log", "a+");
+	char buff[MAX_LEN_STRING];
+
+	for (int i = 0; i < signal_codes_len; i++) {
+		rewind(err_file);
+		memset(buff, 0, sizeof(buff));
+
+		TEST_ASSERT_NOT_NULL(freopen("stderr.log", "w", stderr));
+
+		errno = 0;
+		psignal(signal_codes[i], "Lorem Ipsum");
+		TEST_ASSERT_EQUAL_INT(0, errno);
+
+		fflush(stderr);
+
+		TEST_ASSERT_NOT_NULL(fgets(buff, MAX_LEN_STRING, err_file));
+		TEST_ASSERT_NOT_NULL(strstr(buff, "Lorem Ipsum"));
+		fflush(err_file);
+	}
+
+	fclose(err_file);
+	remove("stderr.log");
+#endif
+}
+
+
+TEST(string_errsign, strsignal_psignal)
+{
+/* Disabled because of #695 issue: https://github.com/phoenix-rtos/phoenix-rtos-project/issues/695 */
+#ifdef __phoenix__
+	TEST_IGNORE_MESSAGE("#695 issue");
+#else
+	FILE *err_file = fopen("stderr.log", "a+");
+	char buff[MAX_LEN_STRING];
+
+	for (int i = 0; i < signal_codes_len; i++) {
+		char *signal_output = strsignal(signal_codes[i]);
+
+		rewind(err_file);
+		memset(buff, 0, sizeof(buff));
+
+		TEST_ASSERT_NOT_NULL(freopen("stderr.log", "w", stderr));
+		psignal(signal_codes[i], "");
```

Comment:
We could also check `NULL` and other edge strings, I think the whole ascii set passed in the string would be sufficient. So we will have four `psignal` cases. We could consider creating the separate test group then and place initialization and deinitialization of `err_file` in setup/teardown.